service/src/main/java/org/cbioportal/service/impl/StudyViewServiceImpl.java

```diff
@@ -1,43 +1,52 @@
 package org.cbioportal.service.impl;
 
-import org.cbioportal.model.GenePanelData;
-import org.cbioportal.model.GenomicDataCount;
-import org.cbioportal.model.MolecularProfile;
-import org.cbioportal.model.MolecularProfileCaseIdentifier;
-import org.cbioportal.service.GenePanelService;
-import org.cbioportal.service.MolecularProfileService;
-import org.cbioportal.service.StudyViewService;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.collections.map.MultiKeyMap;
+import org.cbioportal.model.*;
+import org.cbioportal.model.util.Select;
+import org.cbioportal.service.*;
+import org.cbioportal.service.exception.StudyNotFoundException;
 import org.cbioportal.service.util.MolecularProfileUtil;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
-import java.util.ArrayList;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
+import java.util.function.Function;
 import java.util.stream.Collectors;
 
 @Service
 public class StudyViewServiceImpl implements StudyViewService {
+    private static final List<CNA> CNA_TYPES_AMP_AND_HOMDEL = Collections.unmodifiableList(Arrays.asList(CNA.AMP, CNA.HOMDEL));
     @Autowired
     private MolecularProfileService molecularProfileService;
     @Autowired
     private GenePanelService genePanelService;
     @Autowired
     private MolecularProfileUtil molecularProfileUtil;
+    @Autowired
+    private AlterationCountService alterationCountService;
+    @Autowired
+    private SignificantlyMutatedGeneService significantlyMutatedGeneService;
+    @Autowired
+    private SignificantCopyNumberRegionService significantCopyNumberRegionService;
 
     @Override
     public List<GenomicDataCount> getGenomicDataCounts(List<String> studyIds, List<String> sampleIds) {
-        List<MolecularProfileCaseIdentifier> molecularProfileSampleIdentifiers = molecularProfileService.getMolecularProfileCaseIdentifiers(studyIds, sampleIds);
+        if (CollectionUtils.isEmpty(studyIds) || CollectionUtils.isEmpty(sampleIds)) {
```

Comment:
Shouldn't these checks be &&ed, not ored? I realize in practice, that will never happen, but if this gets used by another service / controller, this current behavior would be incorrect if you just passed a list of study ids.

Reply:
This could happen when all the samples are filtered out. I'll move this empty checks to controller layer